examples/kokkos/view/example_restrict.py

```diff
@@ -0,0 +1,555 @@
+"""
+Memory traffic from the global memory is one of the major bottlenecks of GPUs.
+Therefore, using the :code:`__restrict__` qualifier to hint the compiler that pointers do not
+alias can bring significant gains, such as:
+
+1. Use the ``.CONSTANT`` modifier for the load instruction when the data is read-only.
+2. Reduce redundant memory loads by assuming no aliasing between pointers.
+
+See :cite:label:`nvidia-cuda-restrict-pointers` for more details.
+
+However, depending on the context the :code:`__restrict__` qualifier is used, compilers
+may not be able to take advantage of it.
+
+Case studies
+------------
+
+These case studies examine the SASS code generated by:
+
+.. literalinclude:: ../../../../../examples/kokkos/view/example_restrict.cpp
+    :language: c++
+    :start-at: #define OPERATION
+    :end-at: dst[index] +=
+
+for the following implementations:
+
+1. :py:attr:`Method.GLOBAL_KERNEL`, see :py:meth:`TestSASS.test_global_kernel`
+2. :py:attr:`Method.RESTRICT_RECAST_LAMBDA`, see :py:meth:`TestSASS.test_restrict_recast_lambda`
+3. :py:attr:`Method.RESTRICT_RECAST_LOCAL`, see :py:meth:`TestSASS.test_restrict_recast_local`
+4. :py:attr:`Method.RESTRICT_ACCESSOR`, see :py:meth:`TestSASS.test_restrict_accessor`
+5. :py:attr:`Method.RESTRICT_MEMBER`, see :py:meth:`TestSASS.test_restrict_member`
+6. :py:attr:`Method.RESTRICT_VIEW_MEMORY_TRAIT`, see :py:meth:`TestSASS.test_restrict_view_memory_trait`
+7. :py:attr:`Method.LDG_ACCESSOR`, see :py:meth:`TestSASS.test_ldg_accessor`
+
+These implementations either generate a single sequence of loads/operation/store
+when the compiler figures out pointers do not alias (:py:meth:`TestSASS.match_single`),
+or two sequences otherwise (:py:meth:`TestSASS.match_repeated`).
+When it generates two sequences, the memory traffic is increased significantly, as shown in
+:py:class:`TestNCU`.
+
+Strategies preventing pointer aliasing
+--------------------------------------
+
+The following strategies map to a single sequence, but each has a drawback:
+
+#. :py:attr:`Method.GLOBAL_KERNEL`
+
+   .. caution::
+
+        Writing a :code:`__global__` kernel with :code:`__restrict__` qualified arguments is not portable.
```

Comment:
```suggestion
#. :py:attr:`Method.GLOBAL_KERNEL`

    Use a :code:`__global__` kernel with :code:`__restrict__`
    qualified pointer arguments.
   
   .. caution::

        Writing a :code:`__global__` kernel with :code:`__restrict__` qualified arguments is not portable.
```
(i.e., for each method, we can move the description from later in the code to here)

Reply:
Won't do.